Build fails with CockroachDB 22.1.9

Here's an extract of [one of the stacktraces](https://github.com/hibernate/hibernate-reactive/actions/runs/3272511593/jobs/5383609666#step:6:2716):
```
Gradle Test Executor 7 > org.hibernate.reactive.schema.SchemaUpdateCockroachDBTestBase$IndividuallySchemaUpdateCockroachTestBase > testValidationSucceed PASSED                                                      
                                                                                                                                                                                                                     
Gradle Test Executor 7 > org.hibernate.reactive.schema.SchemaUpdateCockroachDBTestBase$IndividuallySchemaUpdateCockroachTestBase > testUpdate FAILED                                                                 
    java.lang.AssertionError: Not equals : CREATE INDEX i_asimple_avalue_astringvalue ON postgres.public.asimple USING btree (avalue ASC, astringvalue DESC) != CREATE UNIQUE INDEX asimple_pkey ON postgres.public.$
simple USING btree (id ASC)                                                      
```

A branch with Cockroach already updated is here: https://github.com/DavideD/hibernate-reactive/tree/update-cockroach

--- a/README.md
+++ b/README.md
@@ -34,7 +34,7 @@ Hibernate Reactive has been tested with:
 - MySQL 8
 - MariaDB 10
 - Db2 11.5
-- CockroachDB 21.2
+- CockroachDB 22.1
 - MS SQL Server 2019
 - Oracle 21.3
 - [Hibernate ORM][] 5.6.12.Final

--- a/hibernate-reactive-core/src/test/java/org/hibernate/reactive/containers/CockroachDBDatabase.java
+++ b/hibernate-reactive-core/src/test/java/org/hibernate/reactive/containers/CockroachDBDatabase.java
@@ -25,7 +25,7 @@ class CockroachDBDatabase extends PostgreSQLDatabase {
 	 * TIP: To reuse the same containers across multiple runs, set `testcontainers.reuse.enable=true` in a file located
 	 * at `$HOME/.testcontainers.properties` (create the file if it does not exist).
 	 */
-	public static final CockroachContainer cockroachDb = new CockroachContainer( imageName( "cockroachdb/cockroach", "v21.2.14" ) )
+	public static final CockroachContainer cockroachDb = new CockroachContainer( imageName( "cockroachdb/cockroach", "v22.1.9" ) )
 			// Username, password and database are not supported by test container at the moment
 			// Testcontainers will use a database named 'postgres' and the 'root' user
 			.withReuse( true );

--- a/hibernate-reactive-core/src/test/java/org/hibernate/reactive/schema/SchemaUpdateCockroachDBTestBase.java
+++ b/hibernate-reactive-core/src/test/java/org/hibernate/reactive/schema/SchemaUpdateCockroachDBTestBase.java
@@ -31,6 +31,7 @@ import org.junit.Rule;
 import org.junit.Test;
 
 import io.vertx.ext.unit.TestContext;
+import org.assertj.core.api.Assertions;
 
 import static org.hibernate.reactive.containers.DatabaseConfiguration.DBType.COCKROACHDB;
 import static org.hibernate.tool.schema.JdbcMetadaAccessStrategy.GROUPED;
@@ -103,6 +104,8 @@ public abstract class SchemaUpdateCockroachDBTestBase extends BaseReactiveTest {
 
 	@Test
 	public void testUpdate(TestContext context) {
+		// this test checks that the correct indexes and foreign constraints have been created
+		// by looking at the content of the pg_catalog schema using the following native queries.
 		final String indexDefinitionQuery =
 				"select indexdef from pg_indexes where schemaname = 'public' and tablename = ? order by indexname";
 
@@ -154,31 +157,19 @@ public abstract class SchemaUpdateCockroachDBTestBase extends BaseReactiveTest {
 										.thenCompose( v -> s.createNativeQuery( indexDefinitionQuery, String.class )
 												.setParameter( 1, "asimple" )
 												.getResultList()
-												.thenAccept( list -> {
-													context.assertEquals(
-															"CREATE INDEX i_asimple_avalue_astringvalue ON postgres.public.asimple USING btree (avalue ASC, astringvalue DESC)",
-															list.get( 0 )
-													);
-													context.assertEquals(
-															"CREATE INDEX i_asimple_avalue_data ON postgres.public.asimple USING btree (avalue DESC, data ASC)",
-															list.get( 1 )
-													);
-													context.assertEquals(
-															"CREATE UNIQUE INDEX \"primary\" ON postgres.public.asimple USING btree (id ASC)",
-															list.get( 2 )
-													);
-													context.assertEquals(
-															"CREATE UNIQUE INDEX u_asimple_astringvalue ON postgres.public.asimple USING btree (astringvalue ASC)",
-															list.get( 3 )
-													);
-												} )
+												.thenAccept( list -> Assertions.assertThat( list ).containsExactlyInAnyOrder(
+														"CREATE INDEX i_asimple_avalue_astringvalue ON postgres.public.asimple USING btree (avalue ASC, astringvalue DESC)",
+														"CREATE INDEX i_asimple_avalue_data ON postgres.public.asimple USING btree (avalue DESC, data ASC)",
+														"CREATE UNIQUE INDEX asimple_pkey ON postgres.public.asimple USING btree (id ASC)",
+														"CREATE UNIQUE INDEX u_asimple_astringvalue ON postgres.public.asimple USING btree (astringvalue ASC)"
+												) )
 										)
 										.thenCompose( v -> s.createNativeQuery( indexDefinitionQuery, String.class )
 												.setParameter( 1, "aother" )
 												.getSingleResult()
 												.thenAccept( result ->
 																	 context.assertEquals(
-																			 "CREATE UNIQUE INDEX \"primary\" ON postgres.public.aother USING btree (id1 ASC, id2 ASC)",
+																			 "CREATE UNIQUE INDEX aother_pkey ON postgres.public.aother USING btree (id1 ASC, id2 ASC)",
 																			 result
 																	 )
 												)
@@ -188,12 +179,11 @@ public abstract class SchemaUpdateCockroachDBTestBase extends BaseReactiveTest {
 												.getSingleResult()
 												.thenAccept( result ->
 																	 context.assertEquals(
-																			 "CREATE UNIQUE INDEX \"primary\" ON postgres.public.aanother USING btree (id ASC)",
+																			 "CREATE UNIQUE INDEX aanother_pkey ON postgres.public.aanother USING btree (id ASC)",
 																			 result
 																	 )
 												)
 										)
-										// check foreign keys
 										.thenCompose( v -> s.createNativeQuery(
 												foreignKeyDefinitionQuery,
 												String.class

--- a/podman.md
+++ b/podman.md
@@ -120,7 +120,7 @@ configured to run the tests:
 ```
 podman run --rm --name=HibernateTestingCockroachDB \
     --hostname=roachrr1 -p 26257:26257 -p 8080:8080 \
-    cockroachdb/cockroach:v21.2.4 start-single-node --insecure
+    cockroachdb/cockroach:v22.1.9 start-single-node --insecure
 ```
 
 Some of tests needs temporary tables and because this is an experimental feature in

--- a/tooling/jbang/CockroachDBReactiveTest.java.qute
+++ b/tooling/jbang/CockroachDBReactiveTest.java.qute
@@ -70,7 +70,7 @@ public class {baseName} {
 	}
 
     @ClassRule
-    public final static CockroachContainer database = new CockroachContainer( imageName( "cockroachdb", "cockroach", "v21.2.4" ) );
+    public final static CockroachContainer database = new CockroachContainer( imageName( "cockroachdb", "cockroach", "v22.1.9" ) );
 
 	private Mutiny.SessionFactory sessionFactory;
 

--- a/tooling/jbang/ReactiveTest.java
+++ b/tooling/jbang/ReactiveTest.java
@@ -232,7 +232,7 @@ public class ReactiveTest {
 		MYSQL( () -> new MySQLContainer( "mysql:8.0.31" ) ),
 		DB2( () -> new Db2Container( "docker.io/ibmcom/db2:11.5.7.0a" ).acceptLicense() ),
 		MARIADB( () -> new MariaDBContainer( "mariadb:10.9.3" ) ),
-		COCKROACHDB( () -> new CockroachContainer( "cockroachdb/cockroach:v21.2.4" ) );
+		COCKROACHDB( () -> new CockroachContainer( "cockroachdb/cockroach:v22.1.9" ) );
 
 		private final Supplier<JdbcDatabaseContainer<?>> containerSupplier;
 

